src/cpu/arm.rs

```diff
@@ -172,13 +180,46 @@ fn detect_features() -> u32 {
     features
 }
 
+#[cfg(all(target_os = "ios", target_arch = "aarch64"))]
+fn detect_features() -> u32 {
+    let mut features = ARMCAP_STATIC;
+
+    extern "C" {
+        fn sysctlbyname(
+            name: *const u8,
+            oldp: *mut core::ffi::c_void,
+            oldlenp: *mut crate::c::size_t,
+            newp: *mut core::ffi::c_void,
+            newlen: crate::c::size_t,
+        ) -> core::ffi::c_int;
+    }
+
+    let mut value: core::ffi::c_int = 0;
+    let mut len = core::mem::size_of_val(&value);
+    let rc = unsafe {
+        sysctlbyname(
+            "hw.optional.armv8_2_sha512\0".as_ptr(),
+            &mut value as *mut _ as _,
```

Comment:
We don't use `as _` style casts but the clippy job in CI doesn't run on this target. Instead you can add this polyfill to `ring::polyfill`:

```
// TODO(MSRV 1.76): Replace with `core::ptr::from_mut`.
#[inline(always)]
pub fn ptr_from_mut<T: ?Sized>(r: &mut T) -> *mut T {
    <*mut T>::from(r)
}
```
and then use `ptr_from_mut(&mut x).cast::<c_void>()`.